src/main/java/com/nesscomputing/httpserver/HttpConnector.java

```
@@ -57,11 +61,31 @@ public boolean isSecure()
     }
 
     /**
+     * Will be called when jetty should choose a port to update the port information.
+     */
+    void setJettyConnector(final Connector connector)
+    {
+        this.connectorHolder.set(connector);
+    }
+
+    /**
      * Returns the system port for this connector.
      */
     public int getPort()
     {
-        return port;
+        if (port != 0) {
+            return port;
+        }
+        else {
+            final Connector connector = connectorHolder.get();
+            if (connector != null) {
+                Preconditions.checkState(connector.getLocalPort() > 0, "no port was set and the connector is not yet started!");
+                return connector.getLocalPort();
+            }
+            else {
+                return 0;
```

Given that the HttpServer sets this unconditionally, shouldn't this be an error state here, not "silently return a bogus value"?